src/Thermal/RadiativeFB.cpp

```diff
@@ -0,0 +1,308 @@
+  temp_q = params->floatparams["temp_q"];
+
+  runit = simunits->r.outscale * simunits->r.outSI;
+  rsmooth = params->floatparams["r_smooth"] / runit;
+
+  temp_au4 = pow(temp_au, 4.0);
+  temp_exp = -2.0 * temp_q;
+
+  rsmooth2 = pow(rsmooth, 2.0);
+}
+
+//=================================================================================================
+//  DiscHeating::DiscHeating()
+/// DiscHeating class destructor.
+//=================================================================================================
+template <int ndim>
+DiscHeating<ndim>::~DiscHeating()
+{
+
+}
+
+//=================================================================================================
+//  DiscHeating::DiscHeating()
+/// Calculates the ambient temperature for particles assuming a fixed temperature profile.
+/// Depending on the case, the temperature profile may be imposed due to a single or binary
+/// star. This routine should only be used for disc systems where the central object temperature
+/// profile is fixed and does not depend on accretion.
+//=================================================================================================
+template <int ndim>
+FLOAT DiscHeating<ndim>::AmbientTemp
+(Particle<ndim> &part,
+ Sinks<ndim> *sinks) {
+  FLOAT temp = 0.0;
+
+  if (Ncentral <= sinks->Nsink) {
+    for (int i = 0; i < Ncentral; ++i) {
+      SinkParticle<ndim> sink = sinks->sink[i];
+      FLOAT dist = Distance(part.r, sink.star->r, 2); // Midplane distance only
+      temp = temp_au4 * pow((dist * dist + rsmooth2), temp_exp);
+    }
+  }
+  return temp;
+}
+
+//=================================================================================================
+//  SinkHeating::SinkHeating()
+/// SinkHeating class constructor.
+//=================================================================================================
+template <int ndim>
+SinkHeating<ndim>::SinkHeating
+(SimUnits *simunits,
+ Parameters *params,
+ int Ncentral_aux)
+{
+  Ncentral = Ncentral_aux;
+
+  // Set variables from params file
+  f_acc    = params->floatparams["f_acc"];
+  r_star   = params->floatparams["r_star"];
+  r_bdwarf = params->floatparams["r_bdwarf"];
+  r_planet = params->floatparams["r_planet"];
+
+  // Unit conversion, who doesn't love this?
+  DOUBLE num, denom;
+
+  temp_unit = simunits->temp.outscale * simunits->temp.outSI;
+
+  // Calculate Boltzmann constant in code units
+  num       = pow(simunits->r.outscale * simunits->r.outSI, 2.0) *
+              simunits->t.outscale * simunits->t.outSI;
+  denom     = simunits->E.outscale * simunits->E.outSI;
+  rad_const = stefboltz * (num * pow(temp_unit, 4.0)) / denom;
+
+  // Calculate gravitational constant in code units
+  num        = pow(simunits->t.outscale * simunits->t.outSI, 2.0) *
+               simunits->m.outscale * simunits->m.outSI;
+  denom      = pow(simunits->r.outscale * simunits->r.outSI, 3.0);
+  grav_const = G_const * (num / denom);
```

Comment:
But the code units are usually chosen (and have been chosen) so the gravitational constant is 1.0.  So is it still 1.0??  This is probably duplicated work anyway and shouldn't be needed, unless there is some other reason to do it again.

Reply:
You're correct, the grav_const is unnecessary.